### ISSUE TITLE
User profile selection when launching QGIS

### ISSUE THREAD

**Issue by anniinakovalainen** [qgis/QGIS#38762]
**Feature description**
When launching QGIS, it would be cool if QGIS would ask **which user profile QGIS will use when launching**. This would improve the user experience with those who use multiple profiles actively (it's very inconvenient to first launch in wrong user profile and to have to launch another QGIS with another profile and if you happen to missclick, which is always fun, you end up with multiple QGIS windows and then you accidentally close the one with the right user profile goddammit).

But of course the main crowd probably uses only the default user profile. That being said I would suggest that QGIS asks the user profile in launch IF there is more than the one user profile. And if we go even further, this feature could be switched on/off through Settings > Options > General and the user could set a default user profile to use always when QGIS launches (and not the one used last).

**To sum up the requested feature:**
- When launching QGIS, the software asks which user profile to use IF there is more than one user profile
- The profile selection in QGIS launch can be deactivated in settings
- User can set a default user profile to use when launching QGIS

**Comment by DelazJ** [comment]
> it's very inconvenient to first launch in wrong user profile and to have to launch another QGIS with another profile 

@anniinakovalainen you can launch QGIS from command line with profile name as parameter (https://docs.qgis.org/3.10/en/docs/user_manual/introduction/qgis_configuration.html#command-line-and-environment-variables)

**Comment by NathanW2** [comment]
Having a profile selector was originally what I was going to do for this
but went against it to avoid being annoying to users when loading.  One
option is to have a Profile Selector mode like Firefox has I think.  So
that if you give something like --list-profiles as a command line arg it
will show the dialog first and then load the rest of the app.  This would
have to be done really early on in the code as the profile stuff is loaded
very early due to init order etc.

**Comment by anniinakovalainen** [comment]
Yes, I think the selector feature alone would be annoying to the basic users who would only have the default user profile. And maybe even to those who have multiple profiles and uses mainly only one. That is why I think this feature needs the IF-parameter and the option to disable the selector and selecting default user profile. I think that tackles the main inconvenience caused to the basic one profile users and at the same time hugely improves the user experience to those who have multiple user profiles and have to switch between them.

But of course, this is only theoretical rambling and it should be tested by the users.

**Comment by FlyingEagle** [comment]
> * When launching QGIS, the software asks which user profile to use IF there is more than one user profile
> * The profile selection in QGIS launch can be deactivated in settings
> * User can set a default user profile to use when launching QGIS

+1
